src/lib.rs

```diff
@@ -414,7 +414,10 @@ where
     /// # }
     /// ```
     #[inline]
-    pub fn on_retry<F>(self, f: F) -> RetryFuture<MakeFutureT, FutureT, BackoffT, F> {
+    pub fn on_retry<F, OnRetryFut>(self, f: F) -> RetryFuture<MakeFutureT, FutureT, BackoffT, F>
+    where
+        F: for<'a> Fn(u32, Option<Duration>, &E) -> OnRetryFut,
```

Comment:
Do you actually need the `for<'a>`? `'a` doesn't seem to be referenced anywhere 🤔

Reply:
You are absolutely right. Ty!